graphstorm_job.sh

```diff
@@ -1,30 +1,38 @@
 #!/bin/bash
 
-# Description: 
-# This is the entry point script for AWS Batch
-# Batch will execute this file and all the commands in this shell script
-# These commands will be executed in a docker container running on EC2 instance and it's image is present in AWS ECR 
 date
 echo "Args: $@"
 env
 echo "jobId: $AWS_BATCH_JOB_ID"
 echo "jobQueue: $AWS_BATCH_JQ_NAME"
 echo "computeEnvironment: $AWS_BATCH_CE_NAME"
 
-# WORK_DIR=$2
 COMMAND=$1
+SOURCE_REF=$2
+REMOTE=$3
+# WORK_DIR=$2
 # VAR1="chmod +x ./pytest_check.sh && ./pytest_check.sh"
 # VAR2="chmod +x ./lint_check.sh && ./lint_check.sh"
 
+git clone https://github.com/awslabs/graphstorm.git
+
+cd graphstorm
+
+if [ ! -z $REMOTE ]; then
+    git remote set-url origin $REMOTE
+fi
+
+git fetch origin $SOURCE_REF:working
+git checkout working
```

Comment:
Why checkout working?

Reply:
We create a new branch called working, fetch the PR code in that branch and git checkout working switches the current working branch to 'working', which is now pointing to the same commit as the source branch. This allows us to perform subsequent Git operations, such as building or testing the code, on the source branch.

Reply:
But what if the branch name is not working?

Reply:
working is a branch which we create in the github runners after cloning the repository and we checkout code on this 'working' branch and execute tests in this branch